### DISCUSSION TITLE
How to get built artifacts ?

### DISCUSSION THREAD

**Discussion by stuaxo** [nektos/act#1642] [General]
Hi,
   Thanks for `act`, it's incredibly useful.

I was experimenting with act to build https://github.com/fbertola/bgfx-python locally, naturally at the point when it tries to upload the whls it builds with cibuildwhl it fails (using the github upload action).

Is there any way I can retrieve what it has built up to that point, I saw there are some options around an "artifact server", is this the right place to look ?

```
...
[build/Build wheels on ubuntu-20.04-1]   💬  ::debug::Root artifact directory is /home/stu/projects/external/libraries/bgfx-python/wheelhouse
| Starting artifact upload
| For more detailed logs during the artifact upload process, enable step-debugging: https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging
| Artifact name is valid!
[build/Build wheels on ubuntu-20.04-1]   ❗  ::error::Unable to get ACTIONS_RUNTIME_TOKEN env variable
[build/Build wheels on ubuntu-20.04-1]   ❌  Failure - Main Upload artifact
[build/Build wheels on ubuntu-20.04-1] exitcode '1': failure
[build/Build wheels on ubuntu-20.04-1] 🏁  Job failed
Error: Job 'Build wheels on ${{ matrix.os }}' failed
```

**Comment by fatalcoder524** [comment]
I just use upload-artifact@v4 with invalid params, so the job fails, and the Docker container is not cleaned and killed. Later, I go to Docker Desktop -> Containers -> Actions -> View Files and then retrieve the file.

**Comment by Dr-Electron** [comment]
You can just activate the artifacts server by using the `--artifact-server-path` flag and specifying a path to where you want to store the artifacts so that the download action can use it again.
See docs here: https://nektosact.com/usage/index.html#action-artifacts
